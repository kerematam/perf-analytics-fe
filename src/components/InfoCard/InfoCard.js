import React from 'react'
import { Typography } from '@material-ui/core/'

import styles from './InfoCard.module.scss'

const InfoCard = () => (
  <div>
    <Typography>
      The performanceTiming API, a JavaScript API for measuring the loading
      performance of the requested page, is deprecated but supported in all
      browsers. It has been replaced with the performanceNavigationTiming API.
    </Typography>
    <br />
    <Typography>
      The performance timing API provided read only times, in milliseconds(ms),
      describing at what time each point in the page loading process was
      reached. As displayed in the image below, the navigation process goes from
      navigationStart, unloadEventStart, unloadEventEnd, redirectStart,
      redirectEnd, fetchStart, domainLookupStart, domainLookupEnd, connectStart
      , connectEnd, secureConnectionStart, requestStart, responseStart,
      responseEnd, domLoading, domInteractive, domContentLoadedEventStart,
      domContentLoadedEventEnd, domComplete, loadEventStart, and loadEventEnd.
    </Typography>
    <a
      rel="noopener noreferrer"
      className={styles.read_more}
      href="https://developer.mozilla.org/en-US/docs/Web/Performance/Navigation_and_resource_timings#Performance_Timings"
      target="_blank"
    >
      read more
    </a>
  </div>
)

export default InfoCard
