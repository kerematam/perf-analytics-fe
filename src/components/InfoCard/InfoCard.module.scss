.read_more {
  justify-content: flex-end;
  display: flex;
  margin-right: 10px;
}
