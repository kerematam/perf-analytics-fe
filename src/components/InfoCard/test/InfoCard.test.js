import React from 'react'
import renderer from 'react-test-renderer'
import InfoCard from '../index'

const setup = () => {
  const wrapper = renderer.create(<InfoCard />).toJSON()

  return {
    wrapper,
  }
}

describe('Infocard component', () => {
  test('should match with snapshow', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })
})
