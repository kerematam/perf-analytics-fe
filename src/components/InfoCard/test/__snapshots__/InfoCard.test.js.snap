// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infocard component should match with snapshow 1`] = `
<div>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    The performanceTiming API, a JavaScript API for measuring the loading performance of the requested page, is deprecated but supported in all browsers. It has been replaced with the performanceNavigationTiming API.
  </p>
  <br />
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    The performance timing API provided read only times, in milliseconds(ms), describing at what time each point in the page loading process was reached. As displayed in the image below, the navigation process goes from navigationStart, unloadEventStart, unloadEventEnd, redirectStart, redirectEnd, fetchStart, domainLookupStart, domainLookupEnd, connectStart , connectEnd, secureConnectionStart, requestStart, responseStart, responseEnd, domLoading, domInteractive, domContentLoadedEventStart, domContentLoadedEventEnd, domComplete, loadEventStart, and loadEventEnd.
  </p>
  <a
    className="read_more"
    href="https://developer.mozilla.org/en-US/docs/Web/Performance/Navigation_and_resource_timings#Performance_Timings"
    rel="noopener noreferrer"
    target="_blank"
  >
    read more
  </a>
</div>
`;
